src/machine/board_teensy40.go

```diff
@@ -0,0 +1,177 @@
+	D29 = PD31 // [EMC_31]:    SEMC_DATA09      FLEXPWM3_PWMA01  LPUART7_TX       LPSPI1_PCS1     CSI_DATA22            GPIO4_IO31  ENET2_TDATA01          ~                      ~                      ~
+	D30 = PC23 // [EMC_37]:    SEMC_DATA15      XBAR1_IN23       GPT1_COMPARE3    SAI3_MCLK       CSI_DATA16            GPIO3_IO23  USDHC2_WP             ENET2_RX_EN            FLEXCAN3_RX             ~
+	D31 = PC22 // [EMC_36]:    SEMC_DATA14      XBAR1_IN22       GPT1_COMPARE2    SAI3_TX_DATA    CSI_DATA17            GPIO3_IO22  USDHC1_WP             ENET2_RDATA01          FLEXCAN3_TX             ~
+	D32 = PB12 // [B0_12]:     LCD_DATA08       XBAR1_INOUT10    ARM_TRACE_CLK    SAI1_TX_DATA01  FLEXIO2_FLEXIO12      GPIO2_IO12  SRC_BOOT_CFG08        ENET2_TDATA00           ~                      ~
+	D33 = PD7  // [EMC_07]:    SEMC_DATA07      FLEXPWM2_PWMB00  SAI2_MCLK        XBAR1_INOUT09   FLEXIO1_FLEXIO07      GPIO4_IO07   ~                     ~                      ~                      ~
+	D34 = PC15 // [SD_B0_03]:  USDHC1_DATA1     FLEXPWM1_PWMB01  LPUART8_RTS_B    XBAR1_INOUT07   LPSPI1_SDI            GPIO3_IO15  ENET2_RDATA00         SEMC_CLK6               ~                      ~
+	D35 = PC14 // [SD_B0_02]:  USDHC1_DATA0     FLEXPWM1_PWMA01  LPUART8_CTS_B    XBAR1_INOUT06   LPSPI1_SDO            GPIO3_IO14  ENET2_RX_ER           SEMC_CLK5               ~                      ~
+	D36 = PC13 // [SD_B0_01]:  USDHC1_CLK       FLEXPWM1_PWMB00  LPI2C3_SDA       XBAR1_INOUT05   LPSPI1_PCS0           GPIO3_IO13  FLEXSPIB_SS1_B        ENET2_TX_CLK           ENET2_REF_CLK2          ~
+	D37 = PC12 // [SD_B0_00]:  USDHC1_CMD       FLEXPWM1_PWMA00  LPI2C3_SCL       XBAR1_INOUT04   LPSPI1_SCK            GPIO3_IO12  FLEXSPIA_SS1_B        ENET2_TX_EN            SEMC_DQS4               ~
+	D38 = PC17 // [SD_B0_05]:  USDHC1_DATA3     FLEXPWM1_PWMB02  LPUART8_RX       XBAR1_INOUT09   FLEXSPIB_DQS          GPIO3_IO17  CCM_CLKO2             ENET2_RX_EN             ~                      ~
+	D39 = PC16 // [SD_B0_04]:  USDHC1_DATA2     FLEXPWM1_PWMA02  LPUART8_TX       XBAR1_INOUT08   FLEXSPIB_SS0_B        GPIO3_IO16  CCM_CLKO1             ENET2_RDATA01           ~                      ~
+)
+
+// Analog pins
+const (
+	//  = Pin  // Dig | [Pad]      {Ch}
+	A0  = PA18 // D14 | [AD_B1_02] {7}
+	A1  = PA19 // D15 | [AD_B1_03] {8}
+	A2  = PA23 // D16 | [AD_B1_07] {12}
+	A3  = PA22 // D17 | [AD_B1_06] {11}
+	A4  = PA17 // D18 | [AD_B1_01] {6}
+	A5  = PA16 // D19 | [AD_B1_00] {5}
+	A6  = PA26 // D20 | [AD_B1_10] {15}
+	A7  = PA27 // D21 | [AD_B1_11] {0}
+	A8  = PA24 // D22 | [AD_B1_08] {13}
+	A9  = PA25 // D23 | [AD_B1_09] {14}
+	A10 = PA12 // D24 | [AD_B0_12] {1}
+	A11 = PA13 // D25 | [AD_B0_13] {2}
+	A12 = PA30 // D26 | [AD_B1_14] {128+3}
+	A13 = PA31 // D27 | [AD_B1_15] {128+4}
+)
+
+func init() {
+	initLED()
+	initUART()
+	initSPI()
+	initI2C()
+}
```

Comment:
Initialization should happen in the runtime, not in the machine package. Or otherwise, it should happen in the `Configure` call of a particular peripheral.

For example, clock initialization is usually done in the runtime directly. UART initialization is done in the `UART.Configure` method.
In this case, all these functions are empty so there really is no purpose of including them other than making the code longer (and thus harder to read).

Reply:
These were not intended to initialize the peripherals (which *is* done in the `runtime` package, including clock initialization).

Quite a few other boards register interrupts (e.g. UART RX) in the `machine` package's `func init()`. These were I guess placeholders for any similar initialization. But like you see, they are empty in this initial PR, so I can remove them until they are needed.